services/jobs/src/main/java/com/linkedin/openhouse/jobs/services/JobsRegistry.java

```diff
@@ -20,50 +19,54 @@
 import lombok.AllArgsConstructor;
 import lombok.NonNull;
 import org.apache.commons.collections.MapUtils;
+import org.apache.commons.lang3.SerializationUtils;
 
 @AllArgsConstructor(access = AccessLevel.PRIVATE)
 public class JobsRegistry {
-  private String storageUri;
-  private String authTokenPath;
-  private Map<String, JobLaunchConf> jobLaunchDefaultConfByType;
+  private final String storageUri;
+  private final String authTokenPath;
+  private final Map<String, JobLaunchConf> jobLaunchDefaultConfByType;
 
-  public JobLaunchConf createLaunchConf(String jobId, JobConf conf) {
-    final String type = conf.getJobType().name();
+  public JobLaunchConf createLaunchConf(String jobId, JobConf requestConf) {
+    final String type = requestConf.getJobType().name();
     if (!jobLaunchDefaultConfByType.containsKey(type)) {
       throw new JobEngineException(String.format("Job %s is not supported", type));
     }
-    JobLaunchConf defaultConf = jobLaunchDefaultConfByType.get(type);
-    Map<String, String> propsMap = defaultConf.getSparkProperties();
-    /*
-    if properties has authTokenPath, read and set authToken as spark.sql.catalog.openhouse.auth-token
-    in properties
-    */
-    if (authTokenPath != null) {
-      propsMap.put("spark.sql.catalog.openhouse.auth-token", getToken(authTokenPath));
-    }
-    if (MapUtils.isNotEmpty(conf.getExecutionConf())) {
-      populateSparkProps(conf.getExecutionConf(), propsMap);
+    JobLaunchConf extendedRequestConf = createDefaultLaunchConf(requestConf.getJobType());
+    if (MapUtils.isNotEmpty(requestConf.getExecutionConf())) {
+      populateSparkProperties(
```

Comment:
The executionConf should have the execution properties right? That means override properties with maintenance prefix are also included. May be better to rename this method to something like `populateAllSparkProperties` or just `populateAllProperties`.

Reply:
Fixed